### PR TITLE
fix(feeds): Fix multi selection support by forwarding ISelectionInfo impl on wrapping types

### PR DESCRIPTION
**Pull request by dr1rrb**
Linked to https://github.com/unoplatform/uno/issues/10742

## Bugfix
Fix multi selection support by forwarding ISelectionInfo impl on wrapping types

## What is the current behavior?
Multi select is not working properly depending of how the collection is used

## What is the new behavior?
🙃

## PR Checklist
- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Services/SelectionService.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public IReadOnlyList<ItemIndexRange> GetSelectedRanges()
 	/// <inheritdoc />
 	public void SelectRange(ItemIndexRange itemIndexRange)
 	{
-		if (Update((info, removed) => info.Add(removed), new SelectionIndexRange((uint)itemIndexRange.FirstIndex, (uint)itemIndexRange.LastIndex)))
+		if (Update((info, added) => info.Add(added), new SelectionIndexRange((uint)itemIndexRange.FirstIndex, itemIndexRange.Length)))
```
Comment (Xiaoy312) [inline src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Services/SelectionService.cs:53]: SelectionIndexRange could use a static `From(ItemIndexRange)` method or implicit operator

File: src/Uno.Extensions.Reactive/Core/Axes/SelectionInfo.cs [hunk 2]
```diff
@@ -236,4 +236,45 @@ public override string ToString()
...
+			return rightRanges.Count == 0;
+		}
+		else if (count != rightRanges.Count)
+		{
+			return false;
+		}
+
+		for (var i = 0; i < leftRanges.Count; i++)
+		{
+			if (!leftRanges[i].Equals(rightRanges[i]))
```
Comment (Xiaoy312) [inline src/Uno.Extensions.Reactive/Core/Axes/SelectionInfo.cs:272]: do they have to be sequentially equal?

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.Extensions.Reactive/Core/Axes/SelectionInfo.cs (hunk 2 above)
Comment (dr1rrb) [reply src/Uno.Extensions.Reactive/Core/Axes/SelectionInfo.cs:272]: Ctor of `SelectionInfo` is private and we are always ordering them in the right order so sequence will always be the same

File: src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Services/SelectionService.cs (hunk 1 above)
Comment (dr1rrb) [reply src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Services/SelectionService.cs:53]: Yes but no, the `ItemIndexRange` is in the the UI namespace while the `SelectionIndexRange` is in "core" library. Actually I was planning to use `ItemIndexRange` itself but I was forced to create my own one.

**Pull request merged**